jwst/master_background/master_background_step.py

```diff
@@ -118,11 +141,16 @@ def subtract_2d_background(source, background):
     `~jwst.datamodels.DataModel`
         Background subtracted from source.
     """
+
+    log = logging.getLogger(__name__)
+    log.setLevel(logging.DEBUG)
+
     def _subtract_2d_background(model, background):
         result = model.copy()
         # Handle individual NIRSpec FS, NIRSpec MOS
         if isinstance(model, datamodels.MultiSlitModel):
-            for slit, slitbg in zip(result.slits, background.slits):
+            for i, (slit, slitbg) in enumerate(zip(result.slits, background.slits)):
+                log.info('Master background subtraction being done for slit %i', i)
```

Comment:
I like the info this message provides, but I'm not sure this is where we want it in the flow. It's the place where the actual bkg signal is being subtracted from the (multiple) source data, but that's a really quick process of just doing the subtraction. What takes more time is the part that isn't really written yet and will eventually flesh out the If-else block that starts up at line 101. The process of loading or computing all the individual 1-D bkg spectra, combining them into the master 1-D bkg, and then expanding that into the 2-D space of all the input source data instances is what will take some real time and hence that's where it would be nice to have the logging take place. Unfortunately, someone was clever in the way functions like `expand_to_2d` and `subtract_2d_background` were written, in that they're only called once and internally they do all the looping over the multiple source inputs. So the log messages almost need to be imbedded down at that level.

Reply:
Since there can be a lot of slits, you might want to log info only every 20 slits or so.  This is done in `extract_1d` in extract.py around line 2923 (see `log_increment`).  It's messy, though.  You could just check whether `(i + 1) % 20 == 0`, and if so, then log a message.